staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -68,7 +68,7 @@ func currentMigrationRules() map[string]string {
 type ClientConfigLoader interface {
 	ConfigAccess
 	// IsDefaultConfig returns true if the returned config matches the defaults.
-	IsDefaultConfig(*restclient.Config) bool
+	IsDefaultConfig(*restclient.Config) (bool, error)
```

Comment:
is this really needed?
it will be nice to not have to modify a public signature

Reply:
It's not really needed to fix the bug and I can revert it, but then the unit tests will need to be changed for the `in-cluster not checked on non-empty error` test case. 

In the current implementation if the config is invalid, I'm skipping the in cluster check [here](https://github.com/kubernetes/kubernetes/blob/40e35e611983cea2fa7b150871be355fb5558063/staging/src/k8s.io/client-go/tools/clientcmd/merged_client_builder.go#L116). So keeping the signature the same would add a little additional overhead for checking the in-cluster case if the config is invalid.

Reply:
well, it does not seem widely used https://grep.app/search?current=3&q=IsDefaultConfig&filter[lang][0]=Go

so may be is not a big problem